halimahO/Banka
# #166171056 Disable staff & admin can open bank accounts

#### What does this PR do?
Disable staff and admin from opening bank accounts

#### Description of Task to be completed?

- Build out client authentication middleware and ensure it is working

- Ensure the `/api/v1/accounts` POST route is protected from staff and admin

#### How should this be manually tested?

- Using Postman or Insomnia, make a POST request to the route `http://localhost:3000/api/v1/accounts`

- Provide a staff or admin token in the Authorization headers

- Notice that response has a 403 status

#### Any background context you want to provide?
The original design of the app allowed staff and admin the ability to create bank accounts. This is not desirable because these staff and admin are in the position to credit or debit these same accounts they have opened.
This commit ensures only clients can open bank accounts by providing adequate authentication middleware.

#### What are the relevant pivotal tracker stories?
[#166171056](https://www.pivotaltracker.com/story/show/166171056)

#### Screenshots (if appropriate)
![Screenshot (99)](https://user-images.githubusercontent.com/42539315/58135756-9363c680-7c23-11e9-8cbf-fd95be409987.png)
]
